#! /usr/bin/env pythonw shebang causes "[ERROR] No runtime installed that matches w. Try running "py install w"."

**Install source and version**
- [ ] Installed from the Windows Store
- [ ] Installed with the MSIX from python\.org
- [x] Installed with the MSI from python\.org
- [ ] Installed with `winget install 9NQ7512CXL7T`

Version: <!-- for example, 25.0 -->
25.0

<!--
Note that the only WinGet command we support is the one shown above. Any issues encountered when you have installed using a different WinGet command should be reported first at https://github.com/microsoft/winget-pkgs/issues/
-->

**Describe the bug**
Using this minimal script for testing, named `t.pyw`: 
```
#! /usr/bin/env pythonw

import sys
import time
from ctypes import windll

windll.user32.MessageBoxW(None, "Script started", "Hi", 0x10)

print('This goes to stdout', file=sys.stdout)
time.sleep(3)
print('This goes to stderr', file=sys.stderr)
time.sleep(3)

windll.user32.MessageBoxW(None, "Script finished", "Done", 0x10) 
```

If launched using double click it does not do anything. If launched using  `py t.pyw`, the error message `[ERROR] No runtime installed that matches w. Try running "py install w".` is produced. For some reason the shebang is not processed properly, and in fact the shebang `#! /usr/bin/env pyw` works perfectly, don't know why.

**To Reproduce**
Steps to reproduce the behavior:
1. Open a command prompt.
2. Run `py t.pyw`
3. See error

**Expected behavior**
A normal run, just like when the `#! /usr/bin/env pyw` is used, or like the `#! /usr/bin/env python3` shebang with the legacy installer. The real script which triggered this for me was working perfectly with the legacy installer/launcher  and the `#! /usr/bin/env python3` shebang.

--- a/src/manage/scriptutils.py
+++ b/src/manage/scriptutils.py
@@ -19,11 +19,16 @@ def _find_shebang_command(cmd, full_cmd):
     if not sh_cmd.match("*.exe"):
         sh_cmd = sh_cmd.with_name(sh_cmd.name + ".exe")
 
-    is_default = sh_cmd.match("python.exe") or sh_cmd.match("py.exe")
+    is_wdefault = sh_cmd.match("pythonw.exe") or sh_cmd.match("pyw.exe")
+    is_default = is_wdefault or sh_cmd.match("python.exe") or sh_cmd.match("py.exe")
 
     for i in cmd.get_installs():
         if is_default and i.get("default"):
-            return i
+            if is_wdefault:
+                target = [t for t in i.get("run-for", []) if t.get("windowed")]
+                if target:
+                    return {**i, "executable": i["prefix"] / target[0]["target"]}
+            return {**i, "executable": i["prefix"] / i["executable"]}
         for a in i.get("alias", ()):
             if sh_cmd.match(a["name"]):
                 LOGGER.debug("Matched alias %s in %s", a["name"], i["id"])
@@ -35,7 +40,10 @@ def _find_shebang_command(cmd, full_cmd):
             LOGGER.debug("Matched executable %s in %s", i["executable"], i["id"])
             return i
 
-    # Fallback search for 'python<TAG>.exe' shebangs
+    # Fallback search for 'python[w]<TAG>.exe' shebangs
+    if sh_cmd.match("pythonw*.exe"):
+        tag = sh_cmd.name[7:-4]
+        return cmd.get_install_to_run(f"PythonCore/{tag}", windowed=True)
     if sh_cmd.match("python*.exe"):
         tag = sh_cmd.name[6:-4]
         return cmd.get_install_to_run(f"PythonCore/{tag}")

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -159,7 +159,14 @@ class FakeConfig:
     def get_installs(self, *, include_unmanaged=True, set_default=True):
         return self.installs
 
-    def get_install_to_run(self, tag):
+    def get_install_to_run(self, tag, *, windowed=False):
+        if windowed:
+            i = self.get_install_to_run(tag)
+            target = [t for t in i.get("run-for", []) if t.get("windowed")]
+            if target:
+                return {**i, "executable": i["prefix"] / target[0]["target"]}
+            return i
+
         company, _, tag = tag.replace("/", "\\").rpartition("\\")
         return [i for i in self.installs
                 if i["tag"] == tag and (not company or i["company"] == company)][0]

--- a/tests/test_scriptutils.py
+++ b/tests/test_scriptutils.py
@@ -8,6 +8,7 @@ from pathlib import PurePath
 
 from manage.scriptutils import (
     find_install_from_script,
+    _find_shebang_command,
     _read_script,
     NewEncoding,
     _maybe_quote,
@@ -67,6 +68,25 @@ def test_read_shebang(fake_config, tmp_path, script, expect):
         assert expect == actual
     except LookupError:
         assert not expect
+
+
+def test_default_py_shebang(fake_config, tmp_path):
+    inst = _fake_install("1.0", company="PythonCore", prefix=PurePath("C:\\TestRoot"), default=True)
+    inst["run-for"] = [
+        dict(name="python.exe", target=".\\python.exe"),
+        dict(name="pythonw.exe", target=".\\pythonw.exe", windowed=1),
+    ]
+    fake_config.installs[:] = [inst]
+
+    # Finds the install's default executable
+    assert _find_shebang_command(fake_config, "python")["executable"].match("test-binary-1.0.exe")
+    assert _find_shebang_command(fake_config, "py")["executable"].match("test-binary-1.0.exe")
+    assert _find_shebang_command(fake_config, "python1.0")["executable"].match("test-binary-1.0.exe")
+    # Finds the install's run-for executable with windowed=1
+    assert _find_shebang_command(fake_config, "pythonw")["executable"].match("pythonw.exe")
+    assert _find_shebang_command(fake_config, "pyw")["executable"].match("pythonw.exe")
+    assert _find_shebang_command(fake_config, "pythonw1.0")["executable"].match("pythonw.exe")
+
 
 
 @pytest.mark.parametrize("script, expect", [

